crates/sui-json-rpc-types/src/sui_event.rs

```diff
@@ -1,671 +1,173 @@
-    content = "content"
-)]
-pub enum SuiEvent {
-    /// Move-specific event
-    #[serde(rename_all = "camelCase")]
-    MoveEvent {
-        // TODO: What's the best way to serialize this using `AccountAddress::short_str_lossless` ??
-        package_id: ObjectID,
-        transaction_module: String,
-        sender: SuiAddress,
-        type_: String,
-        #[serde(skip_serializing_if = "Option::is_none")]
-        fields: Option<SuiMoveStruct>,
-        #[serde_as(as = "Base64")]
-        #[schemars(with = "Base64")]
-        bcs: Vec<u8>,
-    },
-    /// Module published
-    #[serde(rename_all = "camelCase")]
-    Publish {
-        sender: SuiAddress,
-        package_id: ObjectID,
-        version: SequenceNumber,
-        digest: ObjectDigest,
-    },
-    /// Coin balance changing event
-    #[serde(rename_all = "camelCase")]
-    CoinBalanceChange {
-        package_id: ObjectID,
-        transaction_module: String,
-        sender: SuiAddress,
-        change_type: BalanceChangeType,
-        owner: Owner,
-        coin_type: String,
-        coin_object_id: ObjectID,
-        version: SequenceNumber,
-        amount: i128,
-    },
-    /// Epoch change
-    EpochChange(EpochId),
-    /// New checkpoint
-    Checkpoint(CheckpointSequenceNumber),
-    /// Transfer objects to new address / wrap in another object / coin
-    #[serde(rename_all = "camelCase")]
-    TransferObject {
-        package_id: ObjectID,
-        transaction_module: String,
-        sender: SuiAddress,
-        recipient: Owner,
-        object_type: String,
-        object_id: ObjectID,
-        version: SequenceNumber,
-    },
-    /// Object mutated.
-    #[serde(rename_all = "camelCase")]
-    MutateObject {
-        package_id: ObjectID,
-        transaction_module: String,
-        sender: SuiAddress,
-        object_type: String,
-        object_id: ObjectID,
-        version: SequenceNumber,
-    },
-    /// Delete object
-    #[serde(rename_all = "camelCase")]
-    DeleteObject {
-        package_id: ObjectID,
-        transaction_module: String,
-        sender: SuiAddress,
-        object_id: ObjectID,
-        version: SequenceNumber,
-    },
-    /// New object creation
-    #[serde(rename_all = "camelCase")]
-    NewObject {
-        package_id: ObjectID,
-        transaction_module: String,
-        sender: SuiAddress,
-        recipient: Owner,
-        object_type: String,
-        object_id: ObjectID,
-        version: SequenceNumber,
-    },
+    /// Move package where this event was emitted.
+    pub package_id: ObjectID,
+    #[schemars(with = "String")]
+    #[serde_as(as = "DisplayFromStr")]
+    /// Move module where this event was emitted.
+    pub transaction_module: Identifier,
+    /// Sender's Sui address.
+    pub sender: SuiAddress,
+    #[schemars(with = "String")]
+    #[serde_as(as = "DisplayFromStr")]
+    /// Move event type.
+    pub type_: StructTag,
+    /// Parsed json value of the event
+    pub parsed_json: Value,
```

Comment:
nit: I think we can make this optional(and in separate PR add config options for bcs v.s. parsed_json)